tommath.h

```diff
@@ -148,10 +148,27 @@ typedef struct  {
 /* callback for mp_prime_random, should fill dst with random bytes and return how many read [upto len] */
 typedef int ltm_prime_callback(unsigned char *dst, int len, void *dat);
 
+#if defined(__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__ >= 301)
+#  define MP_DEPRECATED(x) __attribute__((deprecated("replaced by " #x)))
+#  define _MP_DEPRECATED_PRAGMA(s) _Pragma(#s)
+#  define MP_DEPRECATED_PRAGMA(s) _MP_DEPRECATED_PRAGMA(GCC warning s)
+#elif defined(_MSC_VER) && _MSC_VER >= 1500
+#  define MP_DEPRECATED(x) __declspec(deprecated("replaced by " #x))
+#  define MP_DEPRECATED_PRAGMA(s) __pragma(message(s))
+#else
+#  define MP_DEPRECATED
+#  define MP_DEPRECATED_PRAGMA(s)
+#endif
 
-#define USED(m)     ((m)->used)
-#define DIGIT(m, k) ((m)->dp[(k)])
-#define SIGN(m)     ((m)->sign)
+#if defined(__GNUC__) && __GNUC__ >= 3
```

Comment:
probably add a define `HAS_MP_DEPRECATED_PRAGMA` and check for that as here you're missing out mscv

Reply:
you might be looking at the first commit?

Reply:
no, I'm looking at the diff and for `_MSC_VER >= 1500` is `MP_DEPRECATED_PRAGMA` defined but this isn't catched here here.

Reply:
Sorry I don't understand what do you mean. Please clarify.

Reply:
ah ups now I got it

Reply:
Fixed. I should probably go to sleep.

Reply:
me too gn8